src/sctools/test/test_count.py

```diff
@@ -1,44 +1,578 @@
+
+      4) multi-gene alignments: these records have the same tags and query name, though, at least two such
+         records exist that point to different genes. `count.from_sorted_tagged_bam` must ignore such reads.
+
+    Parameters
+    ----------
+    num_cells : int
+        number of real cells
+    max-genes : int
+        maximum number of genes to use to generate synthetic counts
+    annotation_file : str
+        annotation GTF file to look up gene names from
+    gene_expression_rate : float
+        poisson rate at which each gene is expressed
+    rng_seed : int
+        random number generator seed
+
+    Methods
+    -------
+    generate_synthetic_counts_and_alignment_tags
+        generates synthetic test data as described above and returns an instance of SyntheticDataBundle
+    generate_synthetic_bam_and_counts_matrix
+        generates synthetic test data and writes them to disk
+
+    See Also
+    --------
+    count.from_sorted_tagged_bam
+    """
+
+    cell_barcode_tag = 'CB'
+    molecule_barcode_tag = 'UB'
+    gene_name_tag = 'GE'
+    output_prefix = 'synthetic_'
+    bam_output_filename = output_prefix + 'records.bam'
+    count_matrix_output_filename = output_prefix + 'count_matrix.npy'
+    row_index_output_filename = output_prefix + '_row_index.npy'
+    col_index_output_filename = output_prefix + '_col_index.npy'
+
+    def __init__(self,
+                 num_cells: int,
+                 max_genes: int,
+                 annotation_file: str,
+                 gene_expression_rate: float,
+                 rng_seed: int = 777):
+        self.num_cells = num_cells
+        self.gene_expression_rate = gene_expression_rate
+
+        # initialize the random number generator
+        self.rng: np.random.RandomState = np.random.RandomState(seed=rng_seed)
+
+        # generate gene names
+        self.all_gene_names = self._load_gene_names(annotation_file)
+        self.max_genes = max_genes
+        self.num_genes = len(self.all_gene_names)
+        self.to_be_used_gene_indices: List[int] = self.rng.choice(
+            np.arange(0, self.num_genes, dtype=np.int), size=self.max_genes, replace=False).tolist()
+        self.to_be_used_gene_names = [self.all_gene_names[j] for j in self.to_be_used_gene_indices]
+
+    def generate_synthetic_counts_and_alignment_tags(self,
+                                                     num_duplicates: int,
+                                                     num_missing_some_tags: int,
+                                                     num_multiple_gene_alignments: int,
+                                                     max_gene_hits_per_multiple_gene_alignments: int)\
+            -> SyntheticDataBundle:
+
+        # generate count matrix
+        count_matrix: np.ndarray = self._generate_random_count_matrix()
+
+        # generate primary alignment tags that produce count_matrix
+        primary_alignment_record_tags_set, row_index, col_index = \
```

Comment:
instead of a forward slash, I believe you can use an open parenthesis. [PEP8/maximum-line-length](https://www.python.org/dev/peps/pep-0008/?#maximum-line-length) (see bottom of section).

Reply:
Done.